### PR TITLE
Allow users to set custom nginx.conf location

### PR DESCRIPTION
**Pull request by TisVictress**
This PR implements #358. 
I needed to add the nginx.conf path to the layer's `launchEnv` because the user environment variables are not available for exec.d processes.

### CODE REVIEW

File: integration/testdata/custom_conf_app/buildpack.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+nginx:
+  version: "mainline"
```
Comment (robdimsdale) [inline integration/testdata/custom_conf_app/buildpack.yml:3]: Do we need a `buildpack.yml` file at all for this integration test? Support for it is deprecated so we probably don't want to create this file when we make new test fixtures

File: integration/custom_conf_app_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+			container, err = docker.Container.Run.
+				WithEnv(map[string]string{"PORT": "8080"}).
+				WithPublish("8080").
+				Execute(image.ID)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(container).Should(BeAvailable())
+
+			response, err := http.Get(fmt.Sprintf("http://localhost:%s/index.html", container.HostPort("8080")))
```
Comment (robdimsdale) [inline integration/custom_conf_app_test.go:74]: We can use the [`Serve` matcher in occam](https://github.com/paketo-buildpacks/occam/blob/85d6901da66daf207a7ceef437b4336f47b3041f/matchers/serve.go#L63) to make this cleaner and easier to read.

However, we should do this consistently across all tests, so I'd be happy to push that to a separate issue.

File: detect.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ func Detect(versionParser VersionParser) packit.DetectFunc {
 			},
 		}
 
-		_, err := os.Stat(filepath.Join(context.WorkingDir, ConfFile))
+		_, err := os.Stat(getNginxConfLocation(context.WorkingDir))
```
Comment (robdimsdale) [inline detect.go:34]: If you wanted to make this slightly cleaner you could use the [`fs.Exists()` method in packit](https://github.com/paketo-buildpacks/packit/blob/2247967a3f873b178f6fb16c5e6411646ca0882a/fs/exists.go#L9).

No reason to block this PR on that change, though.

File: build_test.go [hunk 4]
```diff
@@ -478,6 +478,95 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
...
+					},
+				},
+				Layers: packit.Layers{Path: layersDir},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result.Launch.Processes[0].Args[len(result.Launch.Processes[0].Args)-1]).To(Equal("/some-absolute-path/nginx.conf"))
+		})
+	})
+
+	context("when BP_NGINX_CONF_LOCATION is not set", func() {
```
Comment (robdimsdale) [inline build_test.go:545]: Is this test adding anything beyond what the default scenario (called: `"it does a build"`) tests? I can't see any functional difference but I could certainly be missing something.

**Review by robdimsdale — CHANGES_REQUESTED** [review]
I think this looks good. I have a couple of non-blocking questions, a couple of style suggestions, and one request to remove the `buildpack.yml` from the newly-created integration test fixture.

Once the `buildpack.yml` file is removed I am 👍

File: integration/testdata/custom_conf_app/buildpack.yml (hunk 1 above)
Comment (TisVictress) [reply integration/testdata/custom_conf_app/buildpack.yml:3]: No we don't need this file. I'll remove it.

File: build_test.go (hunk 4 above)
Comment (TisVictress) [reply build_test.go:545]: Good point, @robdimsdale. I guess it is ultimately the same test. This test only checks that the path is constructed correctly when the custom path is relative; the other test makes other checks including this. But I think I agree that this isn't needed.

**Review by robdimsdale — APPROVED** [review]

File: integration/custom_conf_app_test.go (hunk 2 above)
Comment (TisVictress) [reply integration/custom_conf_app_test.go:74]: @robdimsdale Let's make this change later.

File: detect.go (hunk 3 above)
Comment (TisVictress) [reply detect.go:34]: @robdimsdale  Let's make this change later

**Review by thitch97 — APPROVED** [review]

**Pull request merged**